src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+  avatarClassName = "UserProfilePicture__avatar",
+  avatarStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
+  containerStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
+  reactionPosition = "right",
   user,
 }) => {
-  // @debt some of the redux patterns exist for this, but I don't believe anything actually uses them/calls this at the moment
-  const muteReactions = useSelector((state) => state.room.mute);
-  const { openUserProfileModal } = useProfileModalControls();
-
-  const [pictureUrl, setPictureUrl] = useState("");
-
-  const randomAvatarUrl = (id: string) =>
-    "/avatars/" +
-    RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
-
-  const avatarUrl = useCallback(
-    (id: string, anonMode?: boolean, pictureUrl?: string) => {
-      if (anonMode || !id) {
-        return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-      }
-
-      if (!miniAvatars && pictureUrl) {
-        return setPictureUrl(pictureUrl);
-      }
-
-      if (miniAvatars) {
-        return setPictureUrl(randomAvatarUrl(id));
-      }
-
-      return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-    },
-    [miniAvatars]
-  );
+  const venueId = useVenueId();
 
-  useEffect(() => {
-    avatarUrl(user.id, user.anonMode, user.pictureUrl);
-  }, [avatarUrl, user.anonMode, user.id, user.pictureUrl]);
+  const { openUserProfileModal } = useProfileModalControls();
 
-  const venueId = useVenueId();
+  // TODO: extract this logic into a generic custom reactions component + improve it?
   const reactions = useReactions(venueId);
-
+  // @debt some of the redux patterns exist for this, but I don't believe anything actually uses them/calls this at the moment. Used in MapPartygoersOverlay
+  const muteReactions = useSelector((state) => state.room.mute);
   const typedReaction = reactions ?? [];
-
   const messagesToBand = typedReaction.find(
     (r) => r.reaction === "messageToTheBand" && r.created_by === user.id
   ) as MessageToTheBandReaction | undefined;
 
-  const imageErrorHandler = useCallback(
-    (
-      event: HTMLImageElement | React.SyntheticEvent<HTMLImageElement, Event>
-    ) => {
-      const randomAvatar = randomAvatarUrl(user.id);
-      setPictureUrl(randomAvatar);
-
-      (event as HTMLImageElement).onerror = null;
-      (event as HTMLImageElement).src = randomAvatar;
-    },
-    [user.id]
+  // TODO: I believe we only need this state to support the imageErrorHandler functionality.. can we just remove it?
+  const [pictureUrl, setPictureUrl] = useState(
+    avatarUrl(user.id, user.anonMode, user.pictureUrl, miniAvatars)
   );
+  useEffect(() => {
+    setPictureUrl(
+      avatarUrl(user.id, user.anonMode, user.pictureUrl, miniAvatars)
+    );
+  }, [miniAvatars, user.anonMode, user.id, user.pictureUrl]);
```

Comment:
[TODO] I believe we only need this state to support the `imageErrorHandler` functionality.. can we just remove it?

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1323/commits/db15bd47191b6c10594877933b651616c3375b67 + https://github.com/sparkletown/sparkle/pull/1323/commits/c3a2f885a4b5391c78e677009d55ba47238ab702 + https://github.com/sparkletown/sparkle/pull/1323/commits/f1469b499cbbb22ffcc4acb88790de65a3e1ddd8